### PR TITLE
PE-37:layout provision d configuration page - Base UI setup

### PR DESCRIPTION
**Pull request by ssalunkhe-opennms**
### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-37

### CODE REVIEW

File: ui/src/components/RequisitionDefinitionsLayout.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+
+import { ref } from 'vue'
+import Button from './Common/Button.vue'
+import dataTableDemo from './Common/Demo/DataTableDemo.vue'
+import reqDefinitionFrom from './ReqDefinitionForm.vue'
```
Comment (mikewrosey) [inline ui/src/components/RequisitionDefinitionsLayout.vue:22]: Let's keep the casing the same for all component imports. `DataTableDemo` etc.

File: ui/src/router/index.ts [hunk 2]
```diff
@@ -29,9 +32,26 @@ const router = createRouter({
...
+        {
+          path: '/provisionConfig/threadPools',
+          name: 'thread',
+          component: threadPoolsForm
+        },
+        {
+          path: '/provisionConfig/reqDefinition/edit/:id',
+          name: 'Edit',
+          component: EditNode
+        }
```
Comment (mikewrosey) [inline ui/src/router/index.ts:53]: This works, but child paths don't need the parent in the string. Could change these to just `reqDefinition`, `threadPools`, and `reqDefinition/edit/:id'`

File: ui/src/router/index.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@ import NodeDetails from '@/containers/NodeDetails.vue'
 import Demo from '../components/Common/Demo/Demo.vue'
 import DataTableDemo from '../components/Common/Demo/DataTableDemo.vue'
 import EditNode from '../components/Common/Demo/EditNode.vue'
+import provisionD from '../components/ProvisionD-Config.vue'
+import reqDefForm from '../components/RequisitionDefinitionsLayout.vue'
+import threadPoolsForm from '../components/ThreadPools.vue'
```
Comment (mikewrosey) [inline ui/src/router/index.ts:9]: Change casing here too, `ProvisionD` etc.

File: ui/src/components/ProvisionD-Config.vue [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<template>
```
Comment (mikewrosey) [inline ui/src/components/ProvisionD-Config.vue:1]: I don't know about mixing PascalCase and kebab-case here. Maybe `ProvisionDConfig.vue` will be more consistent with the other filenames.

**Comment by mikewrosey** [comment]
I think for the architecture here, we need a new folder inside `components` called `Configuration` which contains those 4 component files. This will keep things separate/consistent with Node/Demo/Layout etc.

File: ui/src/components/ProvisionD-Config.vue [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <TabMenu :model="TabMenuItem" :activeIndex="activeIndex"/>
+    <router-view></router-view>
+</template>
+<script setup lang="ts">
+
+import TabMenu from 'primevue/tabmenu';
+import { computed, ref } from 'vue';
+
+const activeIndex = ref(0);
```
Comment (mikewrosey) [inline ui/src/components/ProvisionD-Config.vue:10]: Looks like the `activeIndex` is used properly here, but as you mentioned, doesn't work. I don't know why. It's possible this is an issue with this version of PrimeVue. Routing straight to `/provisionConfig/reqDefinition` like you have it seems fine to me though, and this line can be removed.

If you really want the main `/provisionConfig` route to render the ReqDefForm component, however, you could change the routing below to just `/provisionConfig` and the main index routing to use `path: ''` for the first child route.

File: ui/src/components/ProvisionD-Config.vue [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<template>
+    <TabMenu :model="TabMenuItem" :activeIndex="activeIndex"/>
+    <router-view></router-view>
+</template>
+<script setup lang="ts">
+
+import TabMenu from 'primevue/tabmenu';
+import { computed, ref } from 'vue';
```
Comment (mikewrosey) [inline ui/src/components/ProvisionD-Config.vue:8]: computed import unused and can be removed

File: ui/src/components/ProvisionD-Config.vue [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+const activeIndex = ref(0);
+
+const TabMenuItem = ref([
```
Comment (mikewrosey) [inline ui/src/components/ProvisionD-Config.vue:12]: Can remove wrapping this in a ref() as this array will not need reactive updating

File: ui/src/components/RequisitionDefinitionsLayout.vue (hunk 1 above)
Comment (ssalunkhe-opennms) [reply ui/src/components/RequisitionDefinitionsLayout.vue]: Done

File: ui/src/router/index.ts (hunk 2 above)
Comment (ssalunkhe-opennms) [reply ui/src/router/index.ts:53]: Removed parent string from child

File: ui/src/router/index.ts (hunk 3 above)
Comment (ssalunkhe-opennms) [reply ui/src/router/index.ts]: Done

File: ui/src/components/ProvisionD-Config.vue (hunk 4 above)
Comment (ssalunkhe-opennms) [reply ui/src/components/ProvisionD-Config.vue]: Name changed to ProvisionDConfig.vue

File: ui/src/components/ProvisionD-Config.vue (hunk 5 above)
Comment (ssalunkhe-opennms) [reply ui/src/components/ProvisionD-Config.vue]: Done with changes

File: ui/src/components/ProvisionD-Config.vue (hunk 7 above)
Comment (ssalunkhe-opennms) [reply ui/src/components/ProvisionD-Config.vue]: Done

File: ui/src/components/ProvisionD-Config.vue (hunk 6 above)
Comment (ssalunkhe-opennms) [reply ui/src/components/ProvisionD-Config.vue]: Removed the unused import

**Review by mikewrosey — APPROVED** [review]

**Comment by mikewrosey** [comment]
Merging this, as per Sagar's request

**Pull request merged**